core/client/runtime/pom.xml

```diff
@@ -0,0 +1,172 @@
+          <scope>provided</scope>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.curator</groupId>
+          <artifactId>curator-framework</artifactId>
+          <scope>provided</scope>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.curator</groupId>
+          <artifactId>curator-recipes</artifactId>
+          <scope>provided</scope>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-client</artifactId>
+          <scope>provided</scope>
+        </dependency>
+      </dependencies>
+    </profile>
+
+    <!-- build profile for Facebook Presto -->
+    <!-- Profile used to build alluxio client jar used for Facebook Presto. This will be activated by using -Ppresto-->
+    <!-- in the maven command line. -->
+    <profile>
+      <id>presto</id>
+      <!-- We are excluding guava, jackson-databind and jackson-core are in scope provided, as they are provided by Presto runtime. -->
+      <dependencies>
+        <dependency>
+          <groupId>com.fasterxml.jackson.core</groupId>
+          <artifactId>jackson-core</artifactId>
+          <scope>provided</scope>
+        </dependency>
+        <dependency>
+          <groupId>com.fasterxml.jackson.core</groupId>
+          <artifactId>jackson-databind</artifactId>
+          <scope>provided</scope>
+        </dependency>
+        <dependency>
+          <groupId>com.google.guava</groupId>
+          <artifactId>guava</artifactId>
+          <scope>provided</scope>
+        </dependency>
+        <dependency>
+          <groupId>com.google.inject</groupId>
+          <artifactId>guice</artifactId>
+          <scope>provided</scope>
+        </dependency>
+      </dependencies>
+    </profile>
+  </profiles>
+
+  <build>
+    <plugins>
+      <!--In the uber client jar, both proto and thrift are shaded to avoid version conflicts due to external-->
+      <!--dependencies introduced by external applications. The protobuf dependency is shaded in the public client-->
+      <!--jar rather than in the uber client jar, this is because it is also depended by other libraries included in-->
+      <!--the uber jar, thus shading protobuf during the uber jar packaging will over-shade references of protobuf.-->
+      <!--Ideally, the thrift dependency should also be shaded in the public jar in the same way as shading protobuf,-->
+      <!--however, this will break compilation due to existing signatures referencing thrift shared among modules.-->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>uber-jar</id>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <finalName>${project.artifactId}-${project.version}-jar-with-dependencies</finalName>
```

Comment:
do we still use alt name here? by having this name, this uber jar will not be published to maven repository, is it ok? Alternatively, we could let the final name to be `alluxio-core-client-runtime-1.5.0-SNAPSHOT.jar`